numpy/random/pcg64.pyx

```diff
@@ -125,186 +87,34 @@ cdef class PCG64:
-        number generator. All values are returned as unsigned 64-bit
-        values irrespective of the number of bits produced by the PRNG.
-
-        See the class docstring for the number of bits returned.
-        """
-        return random_raw(&self._bitgen, self.lock, size, output)
-
     def _benchmark(self, Py_ssize_t cnt, method=u'uint64'):
         return benchmark(&self._bitgen, self.lock, cnt, method)
 
-    def seed(self, seed=None, inc=0):
-        """
-        seed(seed=None, inc=0)
-
-        Seed the generator.
-
-        This method is called at initialization. It can be called again to
-        re-seed the generator.
-
-        Parameters
-        ----------
-        seed : int, optional
-            Seed for ``PCG64``. Integer between 0 and 2**128-1.
-        inc : int, optional
-            Increment to use for PCG stream. Integer between 0 and 2**128-1.
-
-        Raises
-        ------
-        ValueError
-            If seed values are out of range for the PRNG.
-        """
-        cdef np.ndarray _seed, _inc
-        ub = 2 ** 128
-        if seed is None:
-            try:
-                _seed = <np.ndarray>random_entropy(4)
-            except RuntimeError:
-                _seed = <np.ndarray>random_entropy(4, 'fallback')
-            _seed = <np.ndarray>_seed.view(np.uint64)
-        else:
-            err_msg = 'seed must be a scalar integer between 0 and ' \
-                      '{ub}'.format(ub=ub)
-            if not np.isscalar(seed):
-                raise TypeError(err_msg)
-            if int(seed) != seed:
-                raise TypeError(err_msg)
-            if seed < 0 or seed > ub:
-                raise ValueError(err_msg)
-            _seed = <np.ndarray>np.empty(2, np.uint64)
-            _seed[0] = int(seed) // 2**64
-            _seed[1] = int(seed) % 2**64
-
-        if not np.isscalar(inc):
-            raise TypeError('inc must be a scalar integer between 0 and {ub}'.format(ub=ub))
-        if inc < 0 or inc > ub or int(inc) != inc:
-            raise ValueError('inc must be a scalar integer between 0 and {ub}'.format(ub=ub))
-        _inc = <np.ndarray>np.empty(2, np.uint64)
-        _inc[0] = int(inc) // 2**64
-        _inc[1] = int(inc) % 2**64
-
-        pcg64_set_seed(&self.rng_state,
-                       <uint64_t *>np.PyArray_DATA(_seed),
-                       <uint64_t *>np.PyArray_DATA(_inc))
-        self._reset_state_variables()
-
-    cdef jump_inplace(self, jumps):
-        """
-        Jump state in-place
-
-        Not part of public API
-
-        Parameters
-        ----------
-        jumps : integer, positive
-            Number of times to jump the state of the rng.
-
-        Notes
-        -----
-        The step size is phi-1 when divided by 2**128 where phi is the
-        golden number.
-        """
-        step = 0x9e3779b97f4a7c15f39cc0605cedc834
-        self.advance(step * int(jumps))
-
-    def jumped(self, jumps=1):
```

Comment:
IMO jumping is an alternative method for producing distinct streams should be supported, even it is is not publicized as the preferred method to do this.  It has the one advantage of guaranteeing no overlap, rather than having probability eps of an overlap.

Reply:
Agreed. Algorithms that are capable of efficient jumping should still expose `.jumped()`.

Reply:
restoring